### ISSUE TITLE
Doesn't show runtime errors on client side

### ISSUE THREAD

**Issue by amannn** [developit/nextjs-preact-demo#18]
From https://github.com/vercel/next.js/issues/16357.

I noticed that while runtime errors that affect the server render show up correctly, the ones that are caused during a client side render will unmount the whole app and leave a blank screen.

This can be reproduced by adding this to a component:

```js
useEffect(() => {
  throw new Error('Error');
}, []);
```

Same is true when you have a runtime error directly during the render, but the component only renders on the client side for the first time (e.g. depending on state).

The error isn't printed anywhere and this makes debugging pretty hard.

**Comment by danjebs** [comment]
@amannn I'm also facing the same issue, and I think it is related to prefresh. If I comment out prefresh from my next.config.js (I use withPlugins to compose a bunch of them), then I get runtime errors shown.

It's obviously not ideal to go back to the dark ages before hot reloading, but it's better than the alternative.

**Comment by amannn** [comment]
@danjebs You mean not using this?

```
const withPrefresh = require('@prefresh/next');
```

I just tried that and then changes to component code aren't reflected at all in the browser (not even a hard reload). At least in the most recent version of Next.js. So not really an option for me right now 😕.

**Comment by danjebs** [comment]
I am using 9.5.2 - I don't recall exactly why, I was upgrading to 9.5.3 and found another bug elsewhere (related to either prefresh/date-fns/react-datepicker) which was fixed by downgrading to 9.5.2.

**Comment by developit** [comment]
I believe there is a fix for this that was landed in either Preact 10.5.4 or the latest prefresh (I think!).

Update: now that #21/#22 are merged this should be fixed.

**Comment by danjebs** [comment]
@developit looks like it's working for me on Preact 10.5.4 and prefresh/next 1.3.0
